python/pyspark/ml/torch/distributor.py

```diff
@@ -0,0 +1,297 @@
+        Exception
+            Thrown when the user requires ssl encryption
+        """
+        if not "ssl_conf":
+            raise Exception(
+                "Distributor doesn't have this functionality. Use TorchDistributor instead."
+            )
+        is_ssl_enabled = get_conf_boolean(self.sc, "spark.ssl.enabled", "false")
+        ignore_ssl = get_conf_boolean(self.sc, self.ssl_conf, "false")  # type: ignore
+        if is_ssl_enabled:
+            name = self.__class__.__name__
+            if ignore_ssl:
+                warnings.warn(
+                    f"""
+                    This cluster has TLS encryption enabled;
+                    however, {name} does not
+                    support data encryption in transit.
+                    The Spark configuration
+                    '{self.ssl_conf}' has been set to
+                    'true' to override this
+                    configuration and use {name} anyway. Please
+                    note this will cause model
+                    parameters and possibly training data to
+                    be sent between nodes unencrypted.
+                    """,
+                    RuntimeWarning,
+                )
+                return
+            raise Exception(
+                f"""
+                This cluster has TLS encryption enabled;
+                however, {name} does not support
+                data encryption in transit. To override
+                this configuration and use {name}
+                anyway, you may set '{self.ssl_conf}'
+                to 'true' in the Spark configuration. Please note this
+                will cause model parameters and possibly training
+                data to be sent between nodes unencrypted.
+                """
+            )
+
+
+class TorchDistributor(Distributor):
+    """
+    A class to support distributed training on PyTorch and PyTorch Lightning using PySpark.
+
+    .. versionadded:: 3.4.0
+
+    Examples
+    --------
+
+    Run PyTorch Training locally on GPU (using a PyTorch native function)
+
+    >>> def train(learning_rate):
+    >>>     import torch.distributed
+    >>>     torch.distributed.init_process_group(backend="nccl")
+    >>>     ...
+    >>>     torch.destroy_process_group()
+    >>>     return model # or anything else
+    >>> distributor = TorchDistributor(framework="pytorch",
+                                  num_processes=2,
+                                  local_mode=True,
+                                  use_gpu=True)
+    >>> model = distributor.run(train, 1e-3)
+
+    Run PyTorch Training on GPU (using a file with PyTorch code)
+
+    >>> distributor = TorchDistributor(framework="pytorch",
+                                  num_processes=2,
+                                  local_mode=False,
+                                  use_gpu=True)
+    >>> distributor.run("/path/to/train.py", *args)
+
+    Run PyTorch Lightning Training
```

Comment:
```suggestion
    Run PyTorch Lightning Training

```

Otherwise, documentation rendering is broken.

Reply:
To make it consistent:
```
Run PyTorch Lightning Training on GPU
```